indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -499,14 +506,25 @@ private void syncFromStorage()
     try {
       if (active) {
         final List<Task> newTasks = taskStorage.getActiveTasks();
+        List<Task> addedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(newTasks), Sets.newHashSet(tasks)));
+        List<Task> removedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(tasks), Sets.newHashSet(newTasks)));
+
+        // Clean up removed Tasks
+        for(Task task : removedTasks){
+          shutdown(task.getId());
```

Comment:
Ok I see, you took out the ```clear()```. IMO it would make sense to just do ```taskLockbox.addAll(newTasks)``` and ```tasks.addAll(newTasks)``` here. The removing of tasks from both things can still go through ```manage()```.